### PR TITLE
tests: change covermode to reduce dmctl test execution time

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

dmctl related tests are slow. each test entry costs more than 1s. This is caused by `atomic` covermode is quite expensive.

```
dmctl test cmd: "pause-relay"
execute time: 1217 ms
dmctl test cmd: "pause-relay -w 127.0.0.1:8262 -w 127.0.0.1:8263"
execute time: 1210 ms
dmctl test cmd: "resume-relay wrong_arg"
execute time: 1226 ms
...
```

race test and line cover counts precisely in a parallel way are not essential in `dmctl` test

### What is changed and how it works?

1. change the cover mode from `atomic` to `count` in dmctl test binary build script. This change will reduce the run time of each dmctl test from `12xx ms` to `xx ms`
2. increase MySQL startup wait time
3. refine `purge-relay` test, including the following change
   - disable heartbeat in the test, so the binary log size of upstream is determinate
   - add a failpoint injection to set syncer unit global checkpoint flush interval, in order to update active relay file as soon as possible.

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by amyangfei** [comment]
run-all-tests

**Comment by GregoryIan** [comment]
ci was broken @amyangfei

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: Makefile [hunk 1]
```diff
@@ -119,7 +119,10 @@ integration_test:
 	@which bin/dm-tracer.test
 	tests/run.sh $(CASE)
 
-coverage:
+coverage_fix_cover_mode:
```
Comment (GregoryIan) [inline Makefile]: what's it used for？

Comment (GregoryIan) [reply Makefile]: oh I know, it's very tricky, add a comment

**Comment by GregoryIan** [comment]
LGTM

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**